src/pybind/ceph_volume_client.py

```diff
@@ -1408,15 +1421,26 @@ def put_object(self, pool_name, object_name, data):
         :type data: bytes
         """
         ioctx = self.rados.open_ioctx(pool_name)
+        if version is not None:
+            wop = ioctx.create_write_op()
+
```

Comment:
I would prefer a minimal branching code path for the versioned/normal case. Also, there is a more "pythonic" was of writing this with context manager (as shown below):

```
try:
    with rados.WriteOpCtx(ioctx) as op:
        if version is not None:
            op.assert_version(version)
        op.write_full(data)
        ioctx.operate_write_op(op, object_name)
except rados.OSError:
    # log error here...
    raise
```
With this you do not have to bother about creating/freeing the write op (`WriteOpCtx` is a context manager) -- that's taken care by special functions invoked by context manager (https://github.com/ceph/ceph/blob/master/src/pybind/rados/rados.pyx#L1900)

Reply:
Cool! I was myself wondering how to get rid of the branches. I should've consider using `with`.